util/src/main/java/io/grpc/util/GracefulSwitchLoadBalancer.java

```diff
@@ -90,18 +97,51 @@ public String toString() {
   private LoadBalancer pendingLb = defaultBalancer;
   private ConnectivityState pendingState;
   private SubchannelPicker pendingPicker;
+  private boolean switchToCalled;
 
   private boolean currentLbIsReady;
 
   public GracefulSwitchLoadBalancer(Helper helper) {
     this.helper = checkNotNull(helper, "helper");
   }
 
+  @Override
+  public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
+    if (switchToCalled) {
+      delegate().handleResolvedAddresses(resolvedAddresses);
+      return;
+    }
+    Config config = (Config) resolvedAddresses.getLoadBalancingPolicyConfig();
```

Comment:
I've added some class-level javadoc. See if it addresses this. Note that EDS is unrelated.

Reply:
The javadoc still doesn't make it clear that `ResolvedAddresses.loadBalancingPolicyConfig` MUST be a `Config` object.  I made the EDS comment as an example; it would be natural to take the configuration that XDS provides and just pass it through without wrapping it.

Reply:
That is part of the LB API itself. This is the same behavior for all LBs that use a config, except for PickFirst. This used to be weird, and now it is going to match everything else.

And if someone messes this up, they will find out very quickly.